bumpversion/version_part.py

```diff
@@ -69,7 +69,39 @@ def is_optional(self):
         return self.value == self.config.optional_value
 
     def __format__(self, format_spec):
-        return self.value
+        formatted = None
```

Comment:
So, I'm not sure what kind of `str` formatting people might want to use but most of the format spec deals with numeric types and that's what most of our users use too, what about we simplify the code to just dealing with integers?
```
try:
  val = int(self.value)
except ValueError:
  return self.value
else:
  return int.__format__(val, format_spec)
```

Reply:
Why hasn't this been merged, seems legit fix, currently there is no formatting applied in the serialize string, despite that being mentioned in the README.md, unless I read it wrong:
`This is templated using the [Python Format String Syntax](https://docs.python.org/3/library/string.html#format-string-syntax). `
I could suggest the same fix in a new P.R. if you want.